libs/auth/src/openssl.test.ts

```diff
@@ -164,110 +177,365 @@ test('openssl - provides both stderr and stdout on error', async () => {
   );
 });
 
+test.each<CreateCertInput>([
+  {
+    certKeyInput: {
+      type: 'private',
+      key: { source: 'file', path: '/path/to/private-key-1.pem' },
+    },
+    certSubject: '//',
+    certType: 'cert_authority_cert',
+    expiryInDays: 365,
+    signingCertAuthorityCertPath: '/path/to/cert-authority-cert.pem',
+    signingPrivateKey: { source: 'file', path: '/path/to/private-key-2.pem' },
+  },
+  {
+    certKeyInput: {
+      type: 'public',
+      key: { source: 'inline', content: 'content' },
+    },
+    certSubject: '//',
+    expiryInDays: 365,
+    signingCertAuthorityCertPath: '/path/to/cert-authority-cert.pem',
+    signingPrivateKey: { source: 'tpm' },
+  },
+])('parseCreateCertInput success', (createCertInput) => {
+  expect(parseCreateCertInput(JSON.stringify(createCertInput))).toEqual(
+    createCertInput
+  );
+});
+
+test.each<string>([
+  '',
+  JSON.stringify({}),
+  JSON.stringify({
+    certKeyInput: {
+      type: 'public',
+      key: { source: 'inline', oops: 'oops' },
+    },
+    certSubject: '//',
+    expiryInDays: 365,
+    signingCertAuthorityCertPath: '/path/to/cert-authority-cert.pem',
+    signingPrivateKey: { source: 'tpm' },
+  }),
```

Comment:
Can you add a comment why this test case is expected to error?

Reply:
Rather than adding a comment, I updated the code itself to make it clearer - good nudge! https://github.com/votingworks/vxsuite/pull/3433/commits/c8eaa7de5bcc2cb5327a55903bd15ba0bcbd5544